File: july2020/Day5_hammingDistance.java
package july2020;

import java.util.*;
import java.util.regex.*;


class Solution {
  
    

    public int hammingDistance(int x, int y)
    {
        // using built method  
        return Integer.bitCount(x ^ y); 
    }


    public int hammingDistance2(int x, int y)
    {
        // Brian karnighan's algorithm
        int count = 0;
        int n = x ^ y;
        
        while(n > 0)
        {
            n = n & (n-1);
            count++;
        }
        
        return count;
    }
        



}

public class Day5_hammingDistance{

  public static void main(String[] args) {
    
    Solution obj = new Solution();
    System.out.println(obj.hammingDistance(1, 4));
    
  }


}

File: july2020/Day5_hammingDistance.py
from collections import Counter
class Solution:
  def hammingDistance(self, x: int, y: int) -> int:
      dic = Counter( '{0:b}'.format(x ^ y)  )
      return dic['1']


if __name__ == "__main__":
    obj = Solution()
    print(obj.hammingDistance(1,4))
        